anomaly/anomaly_detector.py

```
@@ -68,26 +72,29 @@ async def _load_model_impl() -> bool:
             "anomaly_detector",
             error_msg="numpy import failed - heuristic mode active",
             fallback_active=True,
-            metadata={"mode": "heuristic", "reason": str(e)}
+            metadata={"mode": "heuristic", "reason": str(e)},
         )
         return False
-    
+
     if os.path.exists(MODEL_PATH):
         with open(MODEL_PATH, "rb") as f:
             _MODEL = pickle.load(f)
```

Ruff (0.14.10)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)